dashboard/lib/services/user.rb

```diff
@@ -33,4 +34,19 @@ def self.assign_auth_option_params(user, params)
       end
     end
   end
+
+  # The user model accepts nested attributes for school_info. On the finish
+  # signup form, users can associate themselves with a school. This method
+  # looks for the school_info attributes in the form params and updates the
+  # User object. The user_school_infos record is created separately, in an
+  # after_save hook in the User model, so it doesn't need to be created here.
+  def self.assign_school_info_params(user, params)
+    # "school_info_attributes"=>{"country"=>"US", "school_type"=>"public", "school_state"=>"Washington", "school_name"=>"Foo School", "school_zip"=>"98021"}
+    params_school_info = params['school_info_attributes']
+    if params_school_info
+      # Disable validation because the school info fields are optional on the registration page
+      params_school_info[:validation_type] = SchoolInfo::VALIDATION_NONE
```

Comment:
is the `preprocess_school_info` step skipped now? [That function](https://github.com/code-dot-org/code-dot-org/blob/356a72ad8184170f48984a78a82c17b25efe3419/dashboard/app/models/user.rb#L364) also sets this validation type to none

Reply:
I was getting validation errors testing locally, but I'll do another pass and see if that function is being run or not

Reply:
Ok just removed that line and it seems to still work fine. The preprocess_school_info function is getting called. Good catch!